### ISSUE TITLE
Tests with Ray are failing

### ISSUE THREAD

**Issue by nrutledge** [mlabs-haskell/plutus-use-cases#63]
On the **neil/lendex-frontend** branch, there are currently 3 tests failing when running the test suite. All are failing due to an unexpected value resulting from the use of Ray.fromInteger.

When a 27 digit integer such as 333333333333333333333333333 is provided as an input, the resulting Ray ends up with a value of 333333333333333333333333333000000000000000000000000000. This seems incorrect and I have written the tests to expect the trailing zeros not to be added after the 27th digit. However, the current behaviour could be by design.

Please confirm with Anton what the correct behaviour for the Ray.fromInteger function is and update the tests/logic accordingly.

**Comment by Benjmhart** [comment]
@nrutledge can you summarize what changes are still on that branch pending the bugfix?

**Comment by anton-k** [comment]
I think Ray can be substituted with Rational if you use plutus version recent enough.
Original purpose of Rays was for ToSchema istance was missing for Rationals but now it's defined it plutus.
So I guess Rays are no longer needed.
it wouuld be better to subtitute them for Rationals

**Comment by anton-k** [comment]
They have same the same interface as Rationals, so substitute should be strightforward

**Comment by mikekeke** [comment]
Looks like this repo still uses `plutus` version w/o `ToSchema` for `Ratio`.
Current `main` branch:
`> :i Ratio
type Ratio :: * -> *
data Ratio a = a PlutusTx.Ratio.:% a
        -- Defined in `PlutusTx.Ratio'
instance Generic (Ratio a) -- Defined in `PlutusTx.Ratio'
instance Show a => Show (Ratio a) -- Defined in `PlutusTx.Ratio'
instance AdditiveGroup (Ratio Integer)
  -- Defined in `PlutusTx.Ratio'
instance AdditiveMonoid (Ratio Integer)
  -- Defined in `PlutusTx.Ratio'
instance AdditiveSemigroup (Ratio Integer)
  -- Defined in `PlutusTx.Ratio'
instance Eq a => Eq (Ratio a) -- Defined in `PlutusTx.Ratio'
instance FromJSON a => FromJSON (Ratio a)`

Current `main` branch of `plutus` monorepo (`cabal repl` run in `plutus-use-cases`):
`> :i Ratio
type Ratio :: * -> *
data Ratio a = a PlutusTx.Ratio.:% a
        -- Defined in ‘PlutusTx.Ratio’
instance Eq a => Eq (Ratio a) -- Defined in ‘PlutusTx.Ratio’
instance Integral a => Ord (Ratio a) -- Defined in ‘PlutusTx.Ratio’
instance Show a => Show (Ratio a) -- Defined in ‘PlutusTx.Ratio’
instance FromJSON a => FromJSON (Ratio a)
  -- Defined in ‘PlutusTx.Ratio’
instance ToJSON a => ToJSON (Ratio a)
  -- Defined in ‘PlutusTx.Ratio’
instance Generic (Ratio a) -- Defined in ‘PlutusTx.Ratio’
instance ToArgument PlutusTx.Ratio.Rational -- Defined in ‘Schema’
instance ToSchema PlutusTx.Ratio.Rational -- Defined in ‘Schema’`

I guess, # 23 needs to be done before that? Can I take it or there are more suitable candidates for this task, @Benjmhart?

**Comment by Benjmhart** [comment]
this is blocked, pending @cstml 's efforts to move us onto a nix build, which should also get us to the latest version of plutus.

**Comment by mikekeke** [comment]
I kinda made transition already from `Ray` to `Rational` in `misha/issue-63-ray-to-rational`, so it it shouldn't take much time to close this one after move to nix build will be completed.

> made transition already 
At least compiler now complains only about missing `ToSchema` instance for `Rational` atm, but I'll recheck everything when `plutus` version will be updated.

**Comment by nrutledge** [comment]
> @nrutledge can you summarize what changes are still on that branch pending the bugfix?

@Benjmhart The summary can be found here: https://github.com/mlabs-haskell/plutus-use-cases/issues/32

**Comment by cstml** [comment]
@nrutledge if you unpin the `shell.nix` by commenting this line https://github.com/mlabs-haskell/plutus-use-cases/blob/d87a58836c6582952f1d875407a70640266eabec/mlabs/nix/default.nix#L5  and run `niv update plutus` you would get nix to pull the latest plutus-repo with its  `nix-shell` (which in theory) has the latest `plutus`. Could be a good intermediary step (if it works) to unblock.

**Comment by nrutledge** [comment]
Good suggestion @cstml. As long is it won't be duplicating efforts to get migrated to the latest plutus version (not sure what breaking API changes need to be dealt with, etc.) then perhaps @mikekeke can give that a try.

**Comment by mikekeke** [comment]
With #88 `Ray` has been removed in favor of built-in `Rational`
